docs/examples/csi_example.md

```diff
@@ -0,0 +1,252 @@
+<h1 align="center">Stateful Application Backup/Restore - MSSQL</h1>
+<h2 align="center">CSI Volume Snapshotting with AWS EBS</h2>
+
+### Prerequisites
+* OADP operator is [installed](/docs/install_olm.md).
+* Create a credentials secret for AWS:
+
+   `oc create secret generic cloud-credentials --namespace openshift-adp --from-file cloud=<CREDENTIALS_FILE_PATH>`
+
+* Make sure your DataProtectionApplication (DPA) CR is similar to below. Note 
+    the `EnableCSI` feature flag and the `csi` default plugin.
+
+    ```
+    apiVersion: oadp.openshift.io/v1alpha1
+    kind: DataProtectionApplication
+    metadata:
+      name: dpa-sample
+    spec:
+      configuration:
+        velero:
+          defaultPlugins:
+          - openshift
+          - aws
+          - csi
+        restic:
+          enable: false
+        featureFlags:
+          - EnableCSI
+      backupLocations:
+        - name: default
+          velero:
+            provider: aws
+            default: true
+            objectStorage:
+              bucket: my-bucket
+              prefix: my-prefix
+            config:
+              region: us-east-1
+              profile: "default"
+            credential:
+              name: cloud-credentials
+              key: cloud
+      volumeSnapshots:
+        - name: default
+          velero:
+            provider: aws
+            config:
+              region: us-west-2
+              profile: "default"
+  
+    ```
+
+    *Note*: Your VSL region should be your cluster's region. 
+
+
+* Install Velero:
+
+    `oc create -n openshift-adp -f config/samples/oadp_v1alpha1_dpa.yaml`
+
+
+<hr style="height:1px;border:none;color:#333;">
+
+### Create a StorageClass and VolumeShapshotClass:
+
+- A `StorageClass` and a `VolumeSnapshotClass` are needed before the Mssql application 
+is created. The app will map to the `StorageClass`, which contains information about the CSI driver. 
+
+- Include a label in `VolumeSnapshotClass` to let 
+Velero know which to use, and set `deletionPolicy` to  `Retain` in order for
+`VolumeSnapshotContent` to remain after the application namespace is deleted.
+
+`oc create -f docs/examples/manifests/mssql/VolumeSnapshotClass.yaml`
+
+```
+apiVersion: v1
+kind: List
+items:
+  - apiVersion: snapshot.storage.k8s.io/v1
+    kind: VolumeSnapshotClass
+    metadata:
+      name: example-snapclass
+      labels:
+        velero.io/csi-volumesnapshot-class: 'true'
+      annotations:
+        snapshot.storage.kubernetes.io/is-default-class: 'true'
+    driver: ebs.csi.aws.com
+    deletionPolicy: Retain
+```
+
+`gp2-csi` comes as a default `StorageClass` with OpenShift clusters. 
+
+`oc get storageclass` 
+
+If this is not found, create a `StorageClass` like below:
+
+```
+apiVersion: storage.k8s.io/v1 
+kind: StorageClass 
+metadata: 
+  name: gp2-csi
+  annotations:
+    storageclass.kubernetes.io/is-default-class: 'true'
+provisioner: ebs.csi.aws.com
+parameters: 
+  type: gp2
+reclaimPolicy: Delete 
```

Comment:
Shouldn't the `reclaimPolicy` be set to `Retain` ?

Reply:
+1 Retain

Reply:
If StorageClass is set to `Retain` I believe upon restore you would end up with two PVs, the original retained one, and a newly restored one with content from snapshot.
The original PV would be dangling and not connected to anything.. either that or new pv would fail to create due to some kind of conflict which I think velero would be ok with and "skip, already exists"

Reply:
Yes what @kaovilai mentioned was my understanding was well. I can verify though.

Reply:
"skip, already exists" when restoring PV is undesired because the data could've diverged from initial snapshot and we wanted stuff from snapshot in the PV.